### ISSUE TITLE
Realtime source: HTTP

### ISSUE THREAD

**Issue by Gleb-ho** [Maproom/qmapshack#435]
### Is your feature request related to a problem? Please describe.
We have a lot of gis data which comes from different sources, and we are gathering it in our self-written application. It takes a lot of time to export and import it to Qmapshack. So it would be helpful to use realtime data source in Qmapshack which polls http server for placemarks.

### Describe the solution you'd like
New realtime data source - polling KML data from HTTP server, would be great.

### Describe alternatives you've considered
Another approach is to write to MySQL server which Qmapshack use. But there is binary data stored, we cant reliably update it. So if Qmapshack's MySQL storage implementation will be refactored to support external updates, and that update process will be documented - problem solved.

### Give a reason why someone else should spend time on that request
We are using Qmapshack in real search and rescue operations and related training. Qmapshack is very helpful in collaborative work on placemarks database due to it's Database functionality. Also Qmapshack has good tools for measurements and some map drawing operations (waypoint radius, waypoint clone and move). But we have to use more then one gis data program because of Qmapshack's limitation in live sources, so it slows down field headquarter's work.

**Comment by kiozen** [comment]
I doubt you will find anyone willing to spent time on that unless you provide more information and prepare as much work as possible.

* What solution are you using to provide the kml data on  the server? Is it just a plain kml file. Is it embedded in a html page. Is there a reference server to test?
* kml offers a huge variety of elements and stuff. To support all it would need a complete library to read kml and some idea what to display. Just saying: "I want kml" is like saying I want unicorns.
* This sounds like quite a proprietary use-case. I doubt anyone will spend time on that unless it's something available to everyone.

**Comment by Gleb-ho** [comment]
- We have exposed simple http endpoint which serves all our actual placemarks data in kml. We use minimal kml syntax, here is an example of response body:
```
<?xml version="1.0" encoding="UTF-8"?>
<kml xmlns="http://www.opengis.net/kml/2.2">
    <Document>
        <Placemark>
            <name>Экипаж 17</name>
            <Point>
                <coordinates>60.333216,56.89996</coordinates>
            </Point>
            <description>2018-07-01 10:15:58 +0000 UTC&#xA;139/Экипаж 17(0012)&#xA;Согласно Приказу № 2/26а  ПСО Экипаж 17 в полном составе находится в базовом лагере.</description>
        </Placemark>
    </Document>
</kml>
```
- Showing placemarks (with their name) on map view is a key feature we need. Ability to click on it and view it's description is also very useful. Custom styles, icons, polygons, regions, lines, etc are optional. If parsing basic kml syntax is a problem - gpx (which is already supported in Qmapshack) is also acceptable.
- This feature can be used to visualise any kml data source, so use-cases are pretty wide, here is some examples of public data sources:
https://earthquake.usgs.gov/earthquakes/feed/v1.0/kml.php
https://www.nhc.noaa.gov/gis/activekml.php

**Comment by kiozen** [comment]
Both of the public examples are quite different approaches. Just because a services uses KML does not make it the same. The one uses a KML to provide URLs to other KMZ that contain data. And the other one provides data a long a timeline. Both make extensive use of KML, therefore a full blown KML implementation is needed. 

And your special interest request is also different to the other two. 

This all does not catch my curiosity therefore I will not spend my time on it. But - as usual - everyone is welcome to submit a PR. Best practice is to read the developer section of the wiki first:

https://github.com/Maproom/qmapshack/wiki/DocMain#developing-qmapshack

**Also very important:**  PRs must be of public interest. You can not add functionality that is just bound to your interest and no one else can use. The reason for this is the effort of maintaining the code. Once the PR is merged into the source code it is up to the project's responsibility to maintain the code and to fix bugs. Usually this is bound on my time and I am not willing to do it unless it has a public use-case everyone can enjoy.

If you can not provide such a public  benefit I recommend maintaining your own fork with the additional functionality. After all the real-time data stuff is quite isolated from the rest. And the OpenSky module is a good example how to do it. In fact that is the reason why it was added.
